storage/engine/mvcc.go

```diff
@@ -1662,18 +1695,28 @@ func mvccEncodeTimestamp(key proto.EncodedKey, timestamp proto.Timestamp) proto.
 // exactly 12 trailing bytes and they're decoded into a timestamp.
 // The decoded key, timestamp and true are returned to indicate the
 // key is for an MVCC versioned value.
-func MVCCDecodeKey(encodedKey proto.EncodedKey) (proto.Key, proto.Timestamp, bool) {
-	tsBytes, key := encoding.MustDecodeBytes(encodedKey, nil)
+func MVCCDecodeKey(encodedKey proto.EncodedKey) (proto.Key, proto.Timestamp, bool, error) {
```

Comment:
I'm not sure that would clarify anything. Yes, there are 4 return values, but their types are all distinct. If you get the order confused the compiler will warn you. If you leave a return value out the compiler will warn you.